crates/ruff_python_formatter/src/other/interpolated_string_element.rs

```diff
@@ -241,32 +197,51 @@ impl Format<PyFormatContext<'_>> for FormatInterpolatedElement<'_> {
                 }
 
                 if let Some(format_spec) = format_spec.as_deref() {
+                    // ```py
+                    // f"{
+                    //     foo
+                    //     # comment 27
+                    //    :test}"
+                    // ```
+                    if comments.has_trailing_own_line(expression) {
+                        soft_line_break().fmt(f)?;
+                    }
+
                     token(":").fmt(f)?;
 
                     for element in &format_spec.elements {
-                        FormatInterpolatedStringElement::new(
-                            element,
-                            self.context.interpolated_string(),
-                        )
-                        .fmt(f)?;
+                        FormatInterpolatedStringElement::new(element, self.context).fmt(f)?;
                     }
-
-                    // These trailing comments can only occur if the format specifier is
-                    // present. For example,
-                    //
-                    // ```python
-                    // f"{
-                    //    x:.3f
-                    //    # comment
-                    // }"
-                    // ```
-                    //
-                    // Any other trailing comments are attached to the expression itself.
-                    trailing_comments(comments.trailing(self.element)).fmt(f)?;
                 }
 
+                // These trailing comments can only occur if the format specifier is
+                // present. For example,
+                //
+                // ```python
+                // f"{
+                //    x:.3f
+                //    # comment
+                // }"
+                // ```
```

Comment:
I think this will also be a syntax error with the recent CPython parser changes? So, then this case can never occur then unless maybe it's a triple-quoted string?

Reply:
Yes, this is correct. I decided to keep supporting this case for now, so that the formatter doesn't mess it up until we make this a syntax error (which will only be in the next release because I want the formatter to fixup all format spec instances with trailing newlines that have no comments.